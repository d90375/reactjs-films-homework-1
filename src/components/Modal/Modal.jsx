import React from 'react';
import PropTypes from 'prop-types';
import Spinner from '../Spinner';

import styles from './Modal.scss';

const Modal = ({
  trailer, removeTrailerInfo, trailerIsLoading, trailerError,
}) => {
  if (trailerIsLoading) {
    return (
      <div className={styles.container}>
        <Spinner />
      </div>
    );
  }

  if (trailerError) {
    return (
      <div className={styles.container}>
        <div className={styles.message}>
          {`Error! ${trailerError.message}`}
        </div>
        <button className={styles.close} type="button" onClick={removeTrailerInfo}>&#215;</button>
      </div>
    );
  }

  if (trailer) {
    return (
      <div className={styles.container}>
        <iframe
          className={styles.iframe}
          title="trailer"
          src={`https://www.youtube.com/embed/${trailer.key}`}
          frameBorder="0"
          allowFullScreen
        />
        <button className={styles.close} type="button" onClick={removeTrailerInfo}>&#215;</button>
      </div>
    );
  }

  return (
    <div className={styles.container}>
      <div className={styles.message}>
        Trailer not found
      </div>
      <button className={styles.close} type="button" onClick={removeTrailerInfo}>&#215;</button>
    </div>
  );
};

Modal.propTypes = {
  trailer: PropTypes.shape({
    key: PropTypes.string,
  }),
  removeTrailerInfo: PropTypes.func.isRequired,
  trailerIsLoading: PropTypes.bool.isRequired,
  trailerError: PropTypes.shape({
    message: PropTypes.string,
  }),
};

Modal.defaultProps = {
  trailer: null,
  trailerError: null,
};

export default Modal;
