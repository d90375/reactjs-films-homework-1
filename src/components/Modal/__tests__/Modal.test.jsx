import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Modal from '../Modal';

describe('Modal render', () => {
  it('Modal renders correctly', () => {
    const mockCallBack = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<Modal removeTrailerInfo={mockCallBack} trailerIsLoading />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('Modal renders correctly', () => {
    const mockCallBack = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<Modal removeTrailerInfo={mockCallBack} trailerIsLoading={false} trailerError={{ message: 'error' }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('Modal renders correctly', () => {
    const mockCallBack = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<Modal removeTrailerInfo={mockCallBack} trailerIsLoading={false} trailer={{ key: 'as123' }} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('Modal renders correctly', () => {
    const mockCallBack = jest.fn();
    const renderer = new ShallowRenderer();
    renderer.render(<Modal removeTrailerInfo={mockCallBack} trailerIsLoading={false} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
