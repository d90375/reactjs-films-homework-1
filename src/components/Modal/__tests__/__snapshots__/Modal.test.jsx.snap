// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render Modal renders correctly when trailer is loading 1`] = `
<div
  className="container"
>
  <Spinner />
</div>
`;

exports[`Modal render Modal renders correctly when trailer loaded 1`] = `
<div
  className="container"
>
  <iframe
    allowFullScreen={true}
    className="iframe"
    frameBorder="0"
    src="https://www.youtube.com/embed/as123"
    title="trailer"
  />
  <button
    className="close"
    onClick={[MockFunction]}
    type="button"
  >
    ×
  </button>
</div>
`;

exports[`Modal render Modal renders correctly when trailer not found 1`] = `
<div
  className="container"
>
  <div
    className="message"
  >
    Trailer not found
  </div>
  <button
    className="close"
    onClick={[MockFunction]}
    type="button"
  >
    ×
  </button>
</div>
`;

exports[`Modal render Modal renders correctly with error message 1`] = `
<div
  className="container"
>
  <div
    className="message"
  >
    Error! error
  </div>
  <button
    className="close"
    onClick={[MockFunction]}
    type="button"
  >
    ×
  </button>
</div>
`;
